src/cli/mod.rs

```diff
@@ -22,5 +26,30 @@ pub trait CommandLineHandler {
     type Arguments: std::fmt::Debug + Args;
 
     /// Handles the request with the provided arguments. Dev should handle the content to print and how
-    async fn handle(arguments: &Self::Arguments) -> anyhow::Result<()>;
+    async fn handle(global: &GlobalParams, arguments: &Self::Arguments) -> anyhow::Result<()>;
+}
+
+#[derive(Debug, Args, Clone)]
+pub struct GlobalParams {
+    #[arg(
+        help = "The toml config file path for IPC Agent, default to ${HOME}/.ipc_agent/config.toml"
+    )]
+    config_path: Option<String>,
+}
+
+impl GlobalParams {
+    pub fn config_path(&self) -> String {
+        self.config_path.clone().unwrap_or_else(|| {
+            let home = match std::env::var("HOME") {
+                Ok(home) => home,
+                Err(_) => String::from(""), // cannot get home, just use current dir
```

Comment:
I don't think this would lead to an accurate value, right?

Reply:
as discussed, should panic!